front/lib/models/assistant/agent.ts

```diff
@@ -228,6 +231,18 @@ AgentConfiguration.belongsTo(AgentGenerationConfiguration, {
   foreignKey: { name: "generationConfigurationId", allowNull: true }, // null = no generation set for this Agent
 });
 
+// NEW -- AgentConfig -> GenerationConfig (1:N)
+AgentConfiguration.hasMany(AgentGenerationConfiguration, {
+  // TODO(@pr) make it non-nullable
+  foreignKey: { name: "agentConfigurationId", allowNull: true },
+  onDelete: "CASCADE",
```

Comment:
Make sure to test agent deletion (downgrade and GDPR)

We def want to remove the CASCADE eventually, you may need it during the double dep situation

Reply:
When we make it non-nullable you don't really have a choice AFAIK, CASCADE is the default.

Reply:
Tested, works